packages/cactus-plugin-ccmodel-hephaestus/src/main/typescript/plugin-ccmodel-hephaestus.ts

```diff
@@ -0,0 +1,911 @@
+      (log) => new Date(log.timestamp).getTime() > lastAggregatedTime,
+    );
+    console.log(logsToAggregate);
+
+    if (logsToAggregate.length === 0) {
+      const finalTime = new Date();
+
+      this.log.debug(
+        `EVAL-${this.className}-AGGREGATE-CCTX-NO_NEW_LOGS:${finalTime.getTime() - startTime.getTime()}`,
+      );
+      return;
+    }
+    logsToAggregate.forEach((eventEntry) => {
+      const ccTxID = eventEntry.caseID;
+      const eventID = eventEntry.receiptID;
+      const latency = (eventEntry.latency as number) || 0;
+      const carbonFootprint = (eventEntry.carbonFootprint as number) || 0;
+      const cost = (eventEntry.cost as number) || 0;
+
+      if (ccTxSet?.has(ccTxID)) {
+        const existingCCTx = ccTxSet.get(ccTxID);
+        const previousEvents = existingCCTx?.processedCrossChainEvents || [];
+        const numberOfCurrentEvents = previousEvents.length + 1;
+        const previousLatency = existingCCTx?.latency || 0;
+        const previousCarbonFootprint = existingCCTx?.carbonFootprint || 0;
+        const previousCost = existingCCTx?.cost || 0;
+        const currentCost = (cost + previousCost) / numberOfCurrentEvents;
+
+        const updatedMetrics = {
+          processedCrossChainEvents: [...previousEvents, eventID],
+          latency: (latency + previousLatency) / numberOfCurrentEvents,
+          carbonFootprint:
+            (carbonFootprint + previousCarbonFootprint) / numberOfCurrentEvents,
+          cost: currentCost,
+          throughput: Number(
+            latency != 0
+              ? ((
+                  1 /
+                  ((latency + previousLatency) / numberOfCurrentEvents)
+                ).toFixed(3) as unknown as number)
+              : 0,
+          ),
+          latestUpdate: lastAggregated,
+        };
+        this.crossChainModel.setCCTxs(ccTxID, updatedMetrics);
+      } else {
+        metrics = {
+          processedCrossChainEvents: [eventID],
+          latency: latency,
+          carbonFootprint: carbonFootprint,
+          cost: cost,
+          throughput: Number(
+            (latency != 0 ? 1 / latency : 0).toFixed(3) as unknown as number,
+          ),
+          latestUpdate: lastAggregated,
+        };
+        this.crossChainModel.setCCTxs(ccTxID, metrics);
+      }
+    });
+    this.crossChainModel.setLastAggregationDate(newAggregationDate);
+    const finalTime = new Date();
+    this.log.debug(
+      `${this.className}-AGGREGATE-CCTX-SUCCESS:${finalTime.getTime() - startTime.getTime()}`,
+    );
+    return;
+  }
+
+  public async persistCrossChainLogCsv(name?: string): Promise<string> {
+    const startTime = new Date();
+    const columns =
+      this.crossChainLog.getCrossChainLogAttributes() as (keyof CrossChainEvent)[];
+    const logName = name
+      ? `${name}.csv`
+      : `hephaestus_log_${startTime.getTime()}.csv`;
+    const csvFolder = path.join(__dirname, "../", "../", "test", "csv");
```

Comment:
@RafaelAPB Depending on this will break the code in production when it is under a different sub-directory path (`./dist/lib/.../`) I recommend making it configurable through constructor parameters so that this issue can be avoided

Reply:
@brunoffmateus can you please address the path issue

Reply:
@RafaelAPB I forgot to mention, but I added the parameter `ccLogsDir: string` that is set in the constructor. When not provided in the plugin options a default path is used.